core/services/ocr3/plugins/ccip_integration_tests/ping_pong_test.go

```diff
@@ -0,0 +1,95 @@
+package ccip_integration_tests
+
+import (
+	"testing"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	gethcommon "github.com/ethereum/go-ethereum/common"
+
+	"github.com/stretchr/testify/require"
+
+	"golang.org/x/exp/maps"
+
+	pp "github.com/smartcontractkit/chainlink/v2/core/gethwrappers/ccip/generated/ping_pong_demo"
+)
+
+/*
+* Test is setting up 3 chains (let's call them A, B, C), each chain deploys and starts 2 ping pong contracts for the other 2.
+* A ---deploy+start---> (pingPongB, pingPongC)
+* B ---deploy+start---> (pingPongA, pingPongC)
+* C ---deploy+start---> (pingPongA, pingPongB)
+* and then checks that each ping pong contract emitted `CCIPSendRequested` event from the expected source to destination.
+* Test fails if any wiring between contracts is not correct.
+ */
+func TestPingPong(t *testing.T) {
+	_, universes := createUniverses(t, 3)
+	pingPongs := initializePingPongContracts(t, universes)
+	for chainID, universe := range universes {
+		for otherChain, pingPong := range pingPongs[chainID] {
+			t.Log("PingPong From: ", chainID, " To: ", otherChain)
+			_, err := pingPong.StartPingPong(universe.owner)
+			require.NoError(t, err)
+			universe.backend.Commit()
+
+			logIter, err := universe.onramp.FilterCCIPSendRequested(&bind.FilterOpts{Start: 0}, nil)
+			require.NoError(t, err)
+			// Iterate until latest event
+			for logIter.Next() {
+			}
+			log := logIter.Event
+			require.Equal(t, log.DestChainSelector, otherChain)
+			require.Equal(t, log.Message.Sender, pingPong.Address())
+			chainPingPongAddr := pingPongs[otherChain][chainID].Address().Bytes()
+			// With chain agnostic addresses we need to pad the address to the correct length if the receiver is zero prefixed
+			paddedAddr := gethcommon.LeftPadBytes(chainPingPongAddr, len(log.Message.Receiver))
+			require.Equal(t, log.Message.Receiver, paddedAddr)
```

Comment:
Is `log.Message.Receiver` the expected receiver or is `paddedAddr`? I think its paddedAddr, so lets change the order of this assertion:

```suggestion
			require.Equal(t, paddedAddr, log.Message.Receiver)
```